### PR TITLE
Add Spark SructType exporter

### PR DESCRIPTION
**Pull request by pierre-monnet**
Add the ability to export Spark StructType.

- [x] Tests pass
- [x] README.md updated
- [x] CHANGELOG.md entry added

### CODE REVIEW

**Comment by simonharrer** [comment]
I forgot to update the CHANGELOG. Can you rebase your changelog entry so that it appears in the unreleased section again, please?

File: README.md [hunk 1]
```diff
@@ -765,6 +766,9 @@ The export function converts the logical data types of the datacontract into the
 if a server is selected via the `--server` option (based on the `type` of that server). If no server is selected, the
 logical data types are exported.
 
+#### Spark
+
+The export function converts the data contract specification into a StructType Spark schema.
```
Comment (simonharrer) [inline README.md:771]: Can you add a link to the spark schema in the docs? So people can go to the according docs for that StructType?

**Comment by simonharrer** [comment]
Thanks for the PR. Looks nice. My question: should the output not be model dependent? So one would have to call the export per model, to get a structtype for that model? Or is it typical to put multiple struct types into a parent dict? I'm raising this as I'm thinking regarding the integration of our CLI tool with other tools, and with this, one would always has to extract a specific structtype again for further processing.

File: README.md (hunk 1 above)
Comment (pierre-monnet) [reply README.md]: I added some details about Spark Data Types

**Comment by pierre-monnet** [comment]
> Thanks for the PR. Looks nice. My question: should the output not be model dependent? So one would have to call the export per model, to get a structtype for that model? Or is it typical to put multiple struct types into a parent dict? I'm raising this as I'm thinking regarding the integration of our CLI tool with other tools, and with this, one would always has to extract a specific structtype again for further processing.

I changed the SparkExporter to return a string "picture" of python code generated.
I keep the function that returns a dict for programmatic usage

File: datacontract/cli.py [hunk 2]
```diff
@@ -211,6 +211,7 @@ class ImportFormat(str, Enum):
     jsonschema = "jsonschema"
     odcs = "odcs"
     unity = "unity"
+    spark = "spark"
```
Comment (jochenchrist) [inline datacontract/cli.py:214]: This importer format seems wrong here (and causes a merge conflict)

Comment (pierre-monnet) [reply datacontract/cli.py]: I resolved conflict.
By the way it was a mistake to put spark in `ImportFormat`

File: README.md [hunk 3]
```diff
@@ -765,6 +766,10 @@ The export function converts the logical data types of the datacontract into the
 if a server is selected via the `--server` option (based on the `type` of that server). If no server is selected, the
 logical data types are exported.
 
+#### Spark
+
+The export function converts the data contract specification into a StructType Spark schema.  
+Spark DataFrame schema is defined as StructType. For more details about Spark Data Types please see [the spark documentation](https://cloud.google.com/docs/authentication/provide-credentials-adc#how-to)
```
Comment (simonharrer) [inline README.md:772]: The link seems wrong

Comment (pierre-monnet) [reply README.md]: Fixed 😶‍🌫️

**Comment by simonharrer** [comment]
Fine for me. We should, however, document, that we basically generate python code here that uses pyspark.

**Comment by pierre-monnet** [comment]
> Fine for me. We should, however, document, that we basically generate python code here that uses pyspark.

I added this detail in the README

**Comment by simonharrer** [comment]
Thanks! :-)

**Pull request merged**